src/test/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojoTest.java

```diff
@@ -87,4 +88,20 @@ public void testMJAVADOC527_DetectLinksRecursion()
         assertFalse( mojo.isValidJavadocLink( "http://javamail.java.net/mailapi/apidocs", false ) );
         assertTrue( mojo.isValidJavadocLink( "http://commons.apache.org/proper/commons-lang/apidocs", false ) );
     }
+
+    public void test_urlEncode()

```

Comment:
Let's try not to make a general URL encoding utility and test through the public API. 

There are better libraries to encode than URLEncoder though I'm not sure which are in our dependency tree already.